src/provider/form-factory-provider.tsx

```diff
@@ -95,12 +99,32 @@ export const FormFactoryProvider: React.FC<FormFactoryProviderProps> = ({
     EncounterFormProcessor: EncounterFormProcessor,
   });
 
+  const vidateAllForms = () => {
```

Comment:
Two concerns:

1. You've a typo in the function name, so rename it to `validateAllForms`
2. Memoize the function by using a `useCallback(...)`